src/controller/CHIPDevice.cpp

```diff
@@ -258,10 +273,38 @@ CHIP_ERROR Device::Deserialize(const SerializedDevice & input)
     return error;
 }
 
+CHIP_ERROR Device::Persist()
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    if (mStorageDelegate != nullptr)
+    {
+        SerializedDevice serialized;
+        Serialize(serialized);
+
+        // TODO: no need to base-64 the serialized values AGAIN
```

Comment:
I think we are still base-64 encoding it twice right? Once because the device supports serializing to a `SerializedDevice` and then again because the Darwin storage delegate converts everything to base-64

Reply:
it was non-obvious to me because 'inner' is not type-defined.

if it would be inner.base64 that would be understandable. However now it is inner and sizeof(inner) so everything sure looks binary to me.

Consider moving inside the serialize.inner reading/writing since this is were we say "this is binary, no need for base64".